src/main/java/org/snomed/snowstorm/core/data/services/AutoMerger.java

```diff
@@ -0,0 +1,542 @@
+            } else if (ComponentType.Relationship.equals(changedComponentType)) {
+                reapplyRelationshipChanges(mapByIdentifier(sourceConcept.getRelationships()), mapByIdentifier(targetConceptOld.getRelationships()), mapByIdentifier(targetConceptNew.getRelationships()), changedComponentIds, mergedConcept);
+            } else if (ComponentType.ClassAxiom.equals(changedComponentType)) {
+                mergedConcept.setClassAxioms(reapplyAxiomsChanges(mapAxiomByIdentifier(sourceConcept.getClassAxioms()), mapAxiomByIdentifier(targetConceptOld.getClassAxioms()), mapAxiomByIdentifier(targetConceptNew.getClassAxioms()), changedComponentIds, mergedConcept));
+            } else if (ComponentType.GciAxiom.equals(changedComponentType)) {
+                mergedConcept.setGciAxioms(reapplyAxiomsChanges(mapAxiomByIdentifier(sourceConcept.getGciAxioms()), mapAxiomByIdentifier(targetConceptOld.getGciAxioms()), mapAxiomByIdentifier(targetConceptNew.getGciAxioms()), changedComponentIds, mergedConcept));
+            } else if (ComponentType.LanguageReferenceSetMember.equals(changedComponentType)) {
+                reapplyLangMemberChanges(mapLangMembersByIdentifier(sourceConcept.getDescriptions()), mapLangMembersByIdentifier(targetConceptOld.getDescriptions()), mapLangMembersByIdentifier(targetConceptNew.getDescriptions()), changedComponentIds, mergedConcept);
+            }
+        }
+
+        return mergedConcept;
+    }
+
+    private void reapplyConceptChanges(Concept sourceConcept, Concept targetConceptNew, Concept targetConceptOld, Set<String> value, Concept mergedConcept) {
+        mergedConcept.setActive(getValueChanged(targetConceptOld.isActive(), targetConceptNew.isActive(), sourceConcept.isActive()));
+        mergedConcept.setModuleId(getValueChanged(targetConceptOld.getModuleId(), targetConceptNew.getModuleId(), sourceConcept.getModuleId()));
+        mergedConcept.setDefinitionStatus(getValueChanged(targetConceptOld.getDefinitionStatus(), targetConceptNew.getDefinitionStatus(), sourceConcept.getDefinitionStatus()));
+        mergedConcept.setReleaseHash(getValueChanged(targetConceptOld.getReleaseHash(), targetConceptNew.getReleaseHash(), sourceConcept.getReleaseHash()));
+        mergedConcept.setReleased(getValueChanged(targetConceptOld.isReleased(), targetConceptNew.isReleased(), sourceConcept.isReleased()));
+
+        mergedConcept.updateEffectiveTime();
+
+        // If changed, ensure component is re-activated unless the RHS had the inactivation.
+        if (mergedConcept.getEffectiveTime() == null && !sourceConcept.isActive() && targetConceptNew.isActive()) {
+            mergedConcept.setActive(true);
+        }
+    }
+
+    private void reapplyDescriptionChanges(Map<String, SnomedComponent<?>> sourceDescriptions, Map<String, SnomedComponent<?>> targetDescriptionsOld, Map<String, SnomedComponent<?>> targetDescriptionsNew, Set<String> changedDescriptionIds, Concept mergedConcept) {
```

Comment:
SnomedComponent<?> should be Description type.

Reply:
@CoderMChu, I've had to use `?` because I am re-using the `mapByIdentifier` method for all components. This method stores a list of components in a map where the key is the component's identifier. 

Shall I create several non-generic `mapByIdentifier` methods or keep as is?

Reply:
I have updated the methods to make better use of generics to prevent casting.